tests/test_java.py

```diff
@@ -564,3 +579,67 @@ def test_java_jattach_async_profiler_log_output(
         )
         # stop
         assert log_records[1].gprofiler_adapter_extra["ap_log"] == ""  # type: ignore
+
+
+@pytest.mark.parametrize(
+    "change_argv0", [pytest.param(True, id="argv0 is java"), pytest.param(False, id="argv0 is not java")]
+)
+def test_java_different_basename(
+    tmp_path: Path,
+    docker_client: DockerClient,
+    application_docker_image: Image,
+    assert_collapsed: AssertInCollapsed,
+    caplog: LogCaptureFixture,
+    change_argv0: bool,
+) -> None:
+    """
+    Tests that we can profile a Java app that runs with non-java "comm", by reading the argv0 instead.
+    """
+    java_notjava_basename = "java-notjava"
+
+    with make_java_profiler(
+        storage_dir=str(tmp_path),
+        duration=1,
+        java_safemode=JAVA_SAFEMODE_ALL,  # explicitly enable, for basename checks
+    ) as profiler:
+        with _application_docker_container(
+            docker_client,
+            application_docker_image,
+            [],
+            [],
+            application_docker_command=[
+                "bash",
+                "-c",
+                f"{'exec -a java ' if change_argv0 else ''}{java_notjava_basename} -jar Fibonacci.jar",
+            ],
+        ) as container:
+            application_pid = container.attrs["State"]["Pid"]
+            profile = snapshot_pid_profile(profiler, application_pid)
+            if change_argv0:
+                # we changed basename - we should run the profiler
+                assert profile.app_metadata is not None
+                assert (
+                    os.path.basename(profile.app_metadata["execfn"])
+                    == os.path.basename(profile.app_metadata["exe"])
+                    == java_notjava_basename
+                )
+                assert_function_in_collapsed(f"{java_notjava_basename};", profile.stacks)
+                assert_collapsed(profile.stacks)
+            else:
+                # we didn't change basename - we should not run the profiler due to a different basename.
```

Comment:
```suggestion
                # we didn't change basename - we should have not run the profiler due to a different basename.
```

Reply:
Maybe `not have run` but I don't think either because the first part of the sentence is past tense, not past perfrect